### DISCUSSION TITLE
Query Parameters for Dynamic Routing

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#15467] [Help]
Rephrasing a question I had earlier.

I understand dynamic routing to some level when we are talking about 'pages/post/[pid].js':
/pages/post/123
/pages/post/abc

I understand that during SSG these two are seen as distinct routes and produce two pages.  And that with fallback = true, if someone hits with a valid:
/pages/post/story1

That the getStaticProps will run and provide data for 'story1' and this will be considered a new distinct route and product a page.

What I am wondering is whether the same URL can use query parameters to create unique dynamic routes.  For instance, can I have something like:
/pages/post?id=abc
/pages/post?id=123

And these be considered unique dynamic routes?

Asking because we have an SPA application with some permalinks and so the main application is at something like /pages/post but we want to be able to SSG some of the permalinks using ?id=.

**Comment by jsartisan** [comment]
@jaybytez Did you find the solution? I am also stuck here.

**Reply by jaybytez** [reply]
No response and it seems like you can’t.  In fact in SSG mode with fallback=true, logging the context/query from server shows no query parameters, even though the client SSG example does.

So we are converting our pages over to paths without query parameters.
